src/utils/metadata/translate.rs

```diff
@@ -87,9 +87,23 @@ impl TranslateMetadata {
         RawFragment::new(expr)
     }
 
-    pub fn gen_sudo_prefix(&self) -> RawFragment {
-        let expr = if self.sudoed { &gen_sudo_prefix() } else { "" };
-        RawFragment::new(expr)
+    pub fn gen_sudo_prefix(&mut self) -> FragmentKind {
+        if self.sudoed {
+            // Generate a unique variable name for the sudo prefix
+            let id = self.gen_value_id();
+            let var_name = format!("__{}_{}", id, "sudo_prefix");
+            
+            // Create the bash condition to detect sudo dynamically (without trailing space)
+            let condition = r#"if [ "$USER" = "root" ] || [ "$(id -u)" = "0" ]; then echo "" ; elif command -v sudo >/dev/null 2>&1; then echo "sudo" ; else echo "" ; fi"#;
```

Comment:
Instead of doing that let's do:
```
__sudo_<GLOBAL_ID>=$([ "$(id -u)" -ne 0 ] && command -v sudo >/dev/null 2>&1 && printf 'sudo ')
```

and then we can later refer this sudo